internal/util/debug/debug.go

```diff
@@ -118,8 +128,22 @@ func Listen(opts *ListenOpts) (*Handler, error) {
 		rw.WriteHeader(http.StatusOK)
 	})
 
-	readyHandler := http.HandlerFunc(func(rw http.ResponseWriter, _ *http.Request) {
-		// TODO https://github.com/FerretDB/FerretDB/issues/4306
+	readyHandler := http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		if opts.Backend == nil {
+			rw.WriteHeader(http.StatusOK)
+			return
+		}
+
+		ctx := conninfo.Ctx(r.Context(), connInfo)
+
+		_, err := opts.Backend.Status(ctx, nil)
+		if err != nil {
+			opts.L.Error("Backend returned error", zap.Error(err))
+			rw.WriteHeader(http.StatusServiceUnavailable)
+
+			return
+		}
```

Comment:
The backend is only provided if new authentication is enabled. If the old one is used, the check just returns straightforward `200`.

Providing backend, allows us to easily call `Status` which properly pings the database.

Reply:
Yeah for backend authentication, it won't be able to ping.